channeldb/invoices.go

```diff
@@ -2305,15 +2305,19 @@ func (d *DB) DeleteInvoice(invoicesToDelete []InvoiceDeleteRef) error {
 				// fetched invoice key matches the one in the
 				// payment address index.
 				key := payAddrIndex.Get(ref.PayAddr[:])
-				if !bytes.Equal(key, invoiceKey) {
-					return fmt.Errorf("unknown invoice " +
-						"in pay addr index")
-				}
-
-				// Delete from the payment address index.
-				err := payAddrIndex.Delete(ref.PayAddr[:])
-				if err != nil {
-					return err
+				if bytes.Equal(key, invoiceKey) {
+					// Delete from the payment address index.
+					// Note that since the payment address
+					// index has been introduced with an
+					// empty migration it may be possible
+					// that the index doesn't have an entry
+					// for this invoice.
+					// ref: https://github.com/lightningnetwork/lnd/pull/4285/commits/cbf71b5452fa1d3036a43309e490787c5f7f08dc#r426368127
+					if err := payAddrIndex.Delete(
```

Comment:
Would this change result in some calls such a `ListInvoices` failing as we fail to delete something from the `payAddr` index? At a quick glance, I think it'll be OK since we iterate based on the add/settle indexes or just the primary invoice sequence number.

Reply:
Fortunately we never look up invoices by `payAddr` only, it's always `payAddr` and `payHash`. Iteration is always by add/settle indexes like you said.